go/host/l1/blockrepository.go

```diff
@@ -0,0 +1,185 @@
+package l1
+
+import (
+	"errors"
+	"fmt"
+	"math/big"
+	"sync/atomic"
+	"time"
+
+	"github.com/ethereum/go-ethereum"
+	gethcommon "github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	gethlog "github.com/ethereum/go-ethereum/log"
+	"github.com/obscuronet/go-obscuro/go/common"
+	"github.com/obscuronet/go-obscuro/go/common/host"
+	"github.com/obscuronet/go-obscuro/go/common/log"
+	"github.com/obscuronet/go-obscuro/go/common/retry"
+	"github.com/obscuronet/go-obscuro/go/ethadapter"
+)
+
+var (
+	// todo (@matt) make this configurable?
+	_timeoutNoBlocks = 30 * time.Second
+	one              = big.NewInt(1)
+	ErrNoNextBlock   = errors.New("no next block")
+)
+
+// Repository is a host service for subscribing to new blocks and looking up L1 data
+type Repository struct {
```

Comment:
I would also create an interface. Just to make the API and responsibilities more obvious.